### DISCUSSION TITLE
Disable warnings for parameter name mismatch in overriden method

### DISCUSSION THREAD

**Discussion by zoltanszocs** [microsoft/pyright#4960] [Q&A] [answered]
Hi!

I am getting some surprising warnings when overriding some methods:

Method "paintEvent" overrides class "QFrame" in an incompatible manner
Parameter 2 name mismatch: base parameter is named "arg__1", override parameter is named "event"

The warning is in fact correct, I just don't want to use the arg__1 as parameter name in my function and I also do not want to disable reportIncompatibleMethodOverride completely.

Is there some solution for this?

I noticed that the docs state that it should not trigger for different parameter names, only for the following: wrong number of parameters, incompatible parameter types, or incompatible return type.

**Comment by erictraut** [comment, marked as answer]
I presume that you have the `reportIncompatibleMethodOverride` check enabled. This is a strict-mode check that is not enabled by default. If you don't want pyright to check for incompatible method overrides, you can simply not enable this check. Or you can leave the check enabled for most of your project but disable it on a per-file basis using a `# pyright: reportIncompatibleMethodOverride=false` comment at the top of the file. Or you can suppress the diagnostic on a per-line basis using `# pyright: ignore[reportIncompatibleMethodOverride]` at the end of a line.

It's fine to use a different parameter name for a positional-only parameter, but for parameters that can be used with keyword arguments, the name of the parameter is significant, and changing the name in your override will make it incompatible with the base class.

**Reply by zoltanszocs** [reply]
Thanks, I missed the issue with keyword arguments.
